maven-core/src/main/java/org/apache/maven/internal/transformation/ConsumerPomArtifactTransformer.java

```diff
@@ -77,13 +82,29 @@ public void injectTransformedArtifacts(MavenProject project, RepositorySystemSes
                 Files.createDirectories(buildDir);
                 generatedFile = Files.createTempFile(buildDir, CONSUMER_POM_CLASSIFIER, "pom");
             }
+            deferDeleteFile(generatedFile);
             project.addAttachedArtifact(new ConsumerPomArtifact(project, generatedFile, session));
         } else if (project.getModel().isRoot()) {
             throw new IllegalStateException(
                     "The use of the root attribute on the model requires the buildconsumer feature to be active");
         }
     }
 
+    private void deferDeleteFile(Path generatedFile) {
+        toDelete.add(generatedFile.toAbsolutePath().toString());
```

Comment:
Why not retain `Path`?

Reply:
I've changed to use `Path` instead, there very few chances we ever hit this code with a custom `FileSystem`.